utils_cv/detection/dataset.py

```diff
@@ -96,38 +150,64 @@ def __init__(
             self.train_ds,
             batch_size=self.batch_size,
             shuffle=True,
-            num_workers=4,
+            num_workers=db_num_workers(),
             collate_fn=collate_fn,
         )
 
         self.test_dl = DataLoader(
             self.test_ds,
             batch_size=self.batch_size,
             shuffle=False,
-            num_workers=4,
+            num_workers=db_num_workers(),
             collate_fn=collate_fn,
         )
 
-    def _get_labels(self) -> List[str]:
+    def _read_annos(self) -> List[str]:
         """ Parses all Pascal VOC formatted annotation files to extract all
         possible labels. """
-        # TODO implement:
-        # - option to bypass this setup set if users already know label names
-        # - option to do 'active' loading so that we preload all annotations
-        # during setup time (instead of lazy loading) - this will let us
-        # do things like get average box size of all the data, etc..
 
+        # If im_dir specified then get image paths and respective annotations
+        # from scanning that directory. Otherwise find all annotation files
+        # and get path to image from annotation.
+        if self.im_dir is None:
+            anno_filenames = sorted(os.listdir(self.root / self.anno_dir))
+            self.im_paths = []
+        else:
```

Comment:
this seems confusing to me. if im_dir is not specified, does that mean `self.anno_dir` is not specified either?

does this not work?
```python
if self.im_dir is None:
    ...
else:
    anno_filenames = sorted(os.listdir(self.root / self.anno_dir))
```

Reply:
In my case I moved .xml annotations to another directory and hence the path to the image is not correct anymore (and hence needs to be ignored when im_dir is specified). I will try to point this out more in the function description. 
And yes, anno_dir always has to be specified.